uitools/cpp/Esri/ArcGISRuntime/Toolkit/SmartLocatorSearchSource.cpp

```diff
@@ -0,0 +1,202 @@
+              {
+                return;
+              }
+
+              if (suggestions()->suggestResults().length() < repeatSuggestResultThreshold())
+              {
+                auto params = suggestions()->suggestParameters();
+                params.setSearchArea(Geometry{});
+                suggestions()->setSuggestParameters(params);
+              }
+            });
+  }
+
+  /*!
+     \brief Destructor.
+   */
+  SmartLocatorSearchSource::~SmartLocatorSearchSource()
+  {
+  }
+
+  /*!
+    \brief Returns the minimum number of results to attempt to retrieve. If there are too few results, the search is repeated with
+    loosened parameters until enough results are accumulated.
+
+    \note If no search is successful, it is still possible to have a total number of results less than this threshold.
+    Set to zero to disable search repeat behavior. Defaults to 1.
+   */
+  int SmartLocatorSearchSource::repeatSearchResultThreshold() const
+  {
+    return m_repeatSearchResultThreshold;
+  }
+
+  /*!
+    \brief Sets the minimum number of results to attempt to retrieve
+    to \a repeatSearchResultThreshold.
+   */
+  void SmartLocatorSearchSource::setRepeatSearchResultThreshold(int repeatSearchResultThreshold)
+  {
+    if (repeatSearchResultThreshold == m_repeatSearchResultThreshold)
+      return;
+
+    m_repeatSearchResultThreshold = repeatSearchResultThreshold;
+    emit repeatSearchResultThresholdChanged();
+  }
+
+  /*!
+    \brief Returns the minimum number of suggestions to attempt to retrieve. If there are too few results, the search is repeated with
+    loosened parameters until enough suggesttions are accumulated.
+
+    \note If no search is successful, it is still possible to have a total number of sugggestions less than this threshold.
+    Set to zero to disable search repeat behavior. Defaults to 6.
+   */
+  int SmartLocatorSearchSource::repeatSuggestResultThreshold() const
+  {
+    return m_repeatSearchSuggestThreshold;
+  }
+
+  /*!
+    \brief Sets the minimum number of suggestions to attempt to retrieve
+    to \a repeatSearchSuggestThreshold.
+   */
+  void SmartLocatorSearchSource::setRepeatSuggestResultThreshold(int repeatSearchSuggestThreshold)
+  {
+    if (repeatSearchSuggestThreshold == m_repeatSearchSuggestThreshold)
+      return;
+
+    m_repeatSearchSuggestThreshold = repeatSearchSuggestThreshold;
+    emit repeatSuggestResultThresholdChanged();
+  }
+
+  /*!
+    \reimp
+   */
+  void SmartLocatorSearchSource::search(const SuggestResult& suggestion, const Geometry area)
+  {
+    // If area is specified, we listen in to the returned results and check to see if the number of
+    // results meets our threshold. If the number of results do not meet our threshold we re-do
+    // the search with the area constraint removed.
+    if (!area.isEmpty() && repeatSearchResultThreshold() > 0)
+    {
+      auto connection = std::make_shared<QMetaObject::Connection>(QMetaObject::Connection{});
```

Comment:
```suggestion
      auto connection = std::make_shared<QMetaObject::Connection>();
```

I think this was using a copy or move constructor.